File: app/src/main/java/de/bitb/astroskop/ui/base/adapter/AdapterPresenter.java
package de.bitb.astroskop.ui.base.adapter;

public interface AdapterPresenter<M> {

    int getAdapterItemCount();

    void onBindAtPosition(IAdapterView<M> holder, int position);

    void onItemClicked(M model);

    void onItemLongClicked(M model);
}

File: app/src/main/java/de/bitb/astroskop/ui/base/adapter/AdapterPresenter.kt
package de.bitb.astroskop.ui.base.adapter

interface AdapterPresenter<M> {

    val adapterItemCount: Int

    fun onBindAtPosition(holder: IAdapterView<M>, position: Int)

    fun onItemClicked(model: M)

    fun onItemLongClicked(model: M)
}
